app/controllers/user_sessions_controller.rb

```diff
@@ -26,8 +26,14 @@ def handle_social_login_flow(auth)
     unless params[:hash_params].to_s.empty?
       hash_params = URI.parse("#" + params[:hash_params]).to_s
     end
-
-    if signed_in?
+    @user = User.where(email: auth["info"]["email"]) ? User.find_by(email: auth["info"]["email"]) : User.find(@identity.user.id)
```

Comment:
Please explain here.

Reply:
It gets the user data either through auth parameters or through omniauth identity and then if it found an existing user, it checks if the user was banned or moderated before signing in. If it didn't find one, that means it is a new user.

Reply:
I have to think here

Reply:
if User.where(email: auth["info"]["email"]) is not present
... this means user is trying to SIGN IN to public lab for the first time. Before account creation @user can't be banned NOR moderated.
Does this makes sense?

Reply:
No I know this is how it works. If it's the first time the user signs in the `@user` variable is nil. Then, the `&`'s in `if @user&.status&.zero?` check whether or not the user is nil and so whether or not the status is nil. It's the same as `if @user and @user.status.zero?` but codeclimate told me to use the `&.` operator. So if `@user` is nil, that whole block is ignored and the function continues, otherwise it checks if it is banned or moderated. I didn't know about it either but learned [here](https://www.rubyguides.com/2018/07/ruby-operators/#The_Safe_Navigator_Operator).

Reply:
Great to see that you learnt about this. I want to tell you that this logic should be inside !signed_in loop